### DISCUSSION TITLE
Recording device keep on switching back to Microphone

### DISCUSSION THREAD

**Discussion by kbiasci** [SamDel/ChromeCast-Desktop-Audio-Streamer#132] [Q&A]
My machine is an ASUS laptop running Windows 11, whenever I run the DAS the recording device goes straight to microphone and cannot be switched to anything else.

It used to work but I guess an update has broken it.
I have tried to remove the rights to the Deskop Apps to access the microphone, this removes the switching but no sound comes out.
The soundcard output is set to wav 44100/16 but nothing comes out.

I am getting a loss here.

Help please?

![image](https://user-images.githubusercontent.com/1813930/172282513-83c9dc47-e13c-4cc7-bbf6-40ecf8cf8a4a.png)
[DAS Log.txt](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/files/8849597/DAS.Log.txt)

**Comment by SamDel** [comment]
Thanks, good info. These lines are in the log:
`ex : [Error initializing the recording device:] IAudioClient::Initialize caused an error: 0x88890008, "Unknown HRESULT".`

That's why some devices are not added to the dropdown.


Here's [some information](https://www.hresult.info/FACILITY_AUDCLNT/0x88890008) about error code 0x88890009.
Can you try [this fix](https://www.vmix.com/knowledgebase/article.aspx/145/error-0x88890008-when-adding-an-audio-input), and let me know if it helped?

**Comment by kbiasci** [comment]
Thanks a lot SamDel, I'll give it a go. 

Something unrelated which seem to work is to enable the "Audio Mixer" device, it seems to combine the sound of the Default Speaker and the Microphone. 
It's very weird. 

Cheers, 

Kevin

**Comment by kbiasci** [comment]
Hmm, sadly this solution had no effect :-(

**Comment by SamDel** [comment]
😞 

According to the error code something goes wrong with the format. ("AUDCLNT_E_UNSUPPORTED_FORMAT" [link](https://docs.microsoft.com/en-us/windows/win32/api/audioclient/nf-audioclient-iaudioclient-initialize))
In [Setup 4.0.2.zip](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/files/8858602/Setup.4.0.2.zip) the device formats the system returns are also logged. Can you try, and post the results?
